### ISSUE TITLE
Bcachefs support

### ISSUE THREAD

**Issue by theoparis** [nix-community/nixos-apple-silicon#232]
I realized that in order to build an ISO with bcachefs support I need to build the ISO using nix, except I can't do that from macOS 😕 
Would it be possible to include bcachefs-tools and the kernel module in the prebuilt ISO? It seems like it should be supported since Linux 6.7... However I don't see it in the kernel config file in this repository despite using the 6.9 kernel.

**Comment by theoparis** [comment]
Actually, I might be able to use the prebuilt ISO with `nix shell`  for bcachefs-tools with an overlay that overrides `fuseSupport`. That way I can try using `bcachefs format` and bcachefs fusemount`. However it'd still be nice if btrfs/bcachefs were supported out of the box.

**Comment by tpwrules** [comment]
This is probably possible, I will have to check that everything cross-compiles properly. We would also have to enable the appropriate kernel config. If you have access to another Linux machine or VM it should be easy to build your own ISO with this feature.

I will consider it for next release, but no promises. Certainly no ZFS!

**Comment by yuyuyureka** [comment]
Adding bcachefs support to a n-a-s installer iso would work the same as adding bcachefs support to the normal NixOS ISO (adding it in supportedFileSystems in iso-configuration/installer-configuration.nix). We have excellent cross-compilation support, and a binary cache populated with all intermediate artifacts, so this is very doable even from an x86_64-linux system.

TL;DR: Assuming the normal NixOS ISOs would not include bcachefs now that it has been removed from the kernel, we are unlikely to include it by default in the n-a-s installer ISO.
